### PR TITLE
Add Github Workflow to check compilation in Windows, Ubuntu and MacOS (latest)

### PR DESCRIPTION
**Pull request by prashanthr05**
As a consequence of adding workflows,
- Updated dependencies
-  Windows compilation was fixed by minor modifications to CMakeLists.txt

Additionally,
- Updated Authors

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+                         -DIDYNTREE_USES_ICUB_MAIN:BOOL=ON \
+                         -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install ..
+            cmake --build . --config ${{matrix.build_type}} --target INSTALL
+                           
+             
+        - name: Source-based Dependencies [Ubuntu/macOS]
+          if: matrix.os == 'ubuntu-latest' || matrix.os == 'macOS-latest'
+          shell: bash
+          run: |
+            # YCM
```
Comment (HosameldinMohamed) [inline .github/workflows/ci.yml]: Just to know, is `YCM` required to build `whole-body-estimators`? If yes maybe we add it to the README?

File: CMakeLists.txt [hunk 2]
```diff
@@ -9,6 +9,12 @@ project(whole-body-estimators LANGUAGES CXX
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+## avoid having M_PI error by importing math symbols from standard cmath
+add_definitions(-D_USE_MATH_DEFINES)
```
Comment (HosameldinMohamed) [inline CMakeLists.txt]: Only for me to understand, could you point me to a line where there's a potential math symbol error?

Comment (prashanthr05) [reply CMakeLists.txt]: If you search for "error" in this [raw log](https://pipelines.actions.githubusercontent.com/YOZBtIxyFZC86xEU2CiDMdyRfOcLH42OneCPmug013AruDqGoK/_apis/pipelines/1/runs/6/signedlogcontent/4?urlExpires=2020-03-16T13%3A55%3A33.1498881Z&urlSigningMethod=HMACV1&urlSignature=aTIwk8On6sVGKYfIRGP4TtZcWL1sDHFE2169iL7y5Do%3D) of that specific job in the workflow, you will find the problem. The raw log can be accessed by pressing `...` in this page https://github.com/prashanthr05/whole-body-estimators/runs/506436067?check_suite_focus=true

This was actually reported by @traversaro  in https://github.com/robotology/whole-body-estimators/issues/6#issuecomment-599040119

Comment (prashanthr05) [reply CMakeLists.txt]: Clicking the raw log points to an expired url. It's better to go directly to the actions tab in the repository and check the status.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (prashanthr05) [reply .github/workflows/ci.yml]: YCM is actually needed for handling some CMake features for YARP and related dependencies.

Comment (prashanthr05) [reply .github/workflows/ci.yml]: I'll update the README

**Comment by prashanthr05** [comment]
> Maybe we add a status badge in the README file.

How did you do that?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (prashanthr05) [reply .github/workflows/ci.yml]: Done in https://github.com/robotology/whole-body-estimators/pull/47/commits/268de81886c387a609e16d11ca40654f6ea6a3de

**Comment by HosameldinMohamed** [comment]
> > Maybe we add a status badge in the README file.
> 
> How did you do that?

I've copied it & modified from another working repo :smile: .

You can modify the link of the badge:
```
![GitHub Actions Status](https://github.com/prashanthr05/whole-body-estimators/workflows/C++%20CI%20Workflow/badge.svg) 
```

- The link of the repository until the directory `workflows` is the location of the `ci.yml` file.
- The phrase `C++%20CI%20Workflow` is the name in the top of the file `ci.yml`:
```
name: C++ CI Workflow
```
- The `badge.svg` stays the same.

**Comment by prashanthr05** [comment]
Then maybe we have to merge to devel and copy the badge of `robotology` remote rather than `prashanthr05` fork?

**Comment by HosameldinMohamed** [comment]
> Then maybe we have to merge to devel and copy the badge of `robotology` remote rather than `prashanthr05` fork?

Yeah, we can do that.

**Comment by fjandrad** [comment]
Sorry for the delay, interesting work I could not have really reviewed it since I do not now these workflow things yet. But looking at the code was useful to start getting acquainted. Nice @prashanthr05 explanations and comments are quite clear.

**Comment by prashanthr05** [comment]
Actually, it was only a very slightly modified version of the workflow in `iDynTree` and `bipedal-locomotion-controllers`.

**Pull request merged**